jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/DockerClientParameters.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.gradle;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Optional;
+
+/**
+ * Object in {@link BuildDockerTask} that configures the Docker executable and the additional
+ * environment variables to use when executing the executable.
+ */
+class DockerClientParameters {
+
+  @Nullable private Path executable;
+  @Nullable private Map<String, String> environment;
+
+  @Input
+  @Nullable
+  @Optional
+  public Path getExecutable() {
+    return executable;
+  }
+
+  public void setExecutable(String executable) {
+    this.executable = Paths.get(executable);
+  }
+
+  @Input
+  @Nullable
+  @Optional
+  public Map<String, String> getEnvironment() {
```

Comment:
I think `setEnvironment` is missing

Reply:
I noticed `build.gradle` works without it (don't know why), but do you think the setter should actually be there?

Reply:
Hmm that's interesting. How does it know what the set.. @loosebazooka Is no setter a valid construct?

Reply:
One thing I also noticed is that, `JibExtension` doesn't have a setter for `useOnlyProjectCache` or `allowInsecureRegistries` either. (And interestingly, the getters for them are not public but package-private.)

But perhaps il'll be just safer to add a setter. Who knows if older Gradle versions won't work if there is no setter. I'll add a setter.

Reply:
I might just be remembering things wrong and perhaps a setter is only necessary if we need to accept a different type than what's declared on the getter.